src/data/set/basic.lean

```diff
@@ -1179,7 +1179,15 @@ ite_same t (s ∩ s') ▸ ite_mono _ (inter_subset_left _ _) (inter_subset_right
 
 lemma ite_inter_inter (t s₁ s₂ s₁' s₂' : set α) :
   t.ite (s₁ ∩ s₂) (s₁' ∩ s₂') = t.ite s₁ s₁' ∩ t.ite s₂ s₂' :=
-by { ext x, finish [set.ite, iff_def] }
+by { ext x, -- `finish` can close this goal
+     simp only [set.ite, set.mem_inter_eq, set.mem_diff, set.mem_union_eq],
+      tidy,
+      { left, exact ⟨‹x ∈ s₁›, ‹x ∈ t›⟩ },
+      { left, exact ⟨‹x ∈ s₂›, ‹x ∈ t›⟩ },
+      { right, exact ⟨‹x ∈ s₁'›, ‹x ∉ t›⟩ },
+      { right, exact ⟨‹x ∈ s₂'›, ‹x ∉ t›⟩ },
+      { left, exact ⟨⟨‹x ∈ s₁›, ‹x ∈ s₂›⟩, ‹x ∈ t›⟩ },
+      { right, exact ⟨⟨‹x ∈ s₁'›, ‹x ∈ s₂'›⟩, ‹x ∉ t›⟩ } }
```

Comment:
```suggestion
begin
  ext x, -- `finish` can close this goal
  simp only [set.ite, set.mem_inter_eq, set.mem_diff, set.mem_union_eq],
  itauto,
end
```
this seems a much better way to me

Reply:
Oh!  Ironically (given the thread that started this whole "finishing `finish`" endeavour) I never even thought to try that!  Thank you.  The proof is now short enough to make it a one-liner.

Reply:
Oh, and just to note it explicitly: this was the only use of `tidy` in the file, which is now eliminated.